src/queryDateCommandHandler.ts

```diff
@@ -0,0 +1,34 @@
+import { Activity, CardFactory, MessageFactory, TurnContext } from "botbuilder";
+import {
+  CommandMessage,
+  TeamsFxBotCommandHandler,
+  TriggerPatterns,
+} from "@microsoft/teamsfx";
+import { AdaptiveCards } from "@microsoft/adaptivecards-tools";
+import { holidaysData } from "./cardData/holidayData";
+import { HolidayCardData } from "./cardModels";
+import holidayTemplate from "./adaptiveCards/notification-holiday.json";
+
+export class QueryDateCommandHandler implements TeamsFxBotCommandHandler {
+  triggerPatterns: TriggerPatterns = /^Query date (.*?)$/i;
+
+  async handleCommandReceived(
+    context: TurnContext,
+    message: CommandMessage
+  ): Promise<string | Partial<Activity>> {
+    // verify the command arguments which are received from the client if needed.
+    console.log(`App received message: ${message.text}`);
+    const queryDateString: string = message.matches[1];
+    console.log(`queryDate: ${queryDateString}`);
+
+    const queryDate = new Date(queryDateString);
+    for (const holiday of holidaysData) {
+        const holidayDate = Date.parse(`${queryDate.getFullYear()}-${holiday.holidayDate}`);
+        if (queryDate.getTime() === holidayDate) {
```

Comment:
Does this work since user input query date is by default calculated as UTC+0 time while the holiday date has specific time zone?

Reply:
Update to use re to match month and day from holiday date.
I tryied to get time zone from `Date` class, but it seems that it will be save as UTC time when convert string to Date. There is no time zone property in `Date` class.